src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -7,14 +7,15 @@ import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.DraftReferral
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
 import java.util.UUID
 
 @RestController
 class ReferralController(private val repository: ReferralRepository) {
   @PostMapping("/draft-referral")
-  fun createDraftReferral(): ResponseEntity<Referral> {
+  fun createDraftReferral(): ResponseEntity<DraftReferral> {
     val referral = Referral()
```

Comment:
`Referral` might be ambiguous soon when we end up having this, the draft and the sent referrals; would it be worth to call this a `ReferralInput`?

Reply:
i think internally the referral entity will remain the singular entity for storing all these type of referral, and the DTO will represent how the referral is presented to the API consumer. what do you think ?

Reply:
Right, I was thinking of something else, `Referral` is the persistence entity. No "Input" postfix needed 😄 

Model-wise, I think it depends.

1. If each entity is to be without a `state` column, then we may have a `sent_referrals` table that references a `draft_referral_id` and some additional fields like `sender`, `sent_at`, etc.

    That kind of structure would help us avoid having unrelated fields in persistence entities (and the unnecessary branching that comes with state/extra fields/validations).

1. On the other hand, `state` entities are simpler but require some extra semantics checks after deserialisation (is this entity a draft or sent referral?)

Reply:
I was gravitating towards (1) because the state is explicit in the code and it's more concise at the database layer. That said, I don't have more than just a gut feeling.

Reply:
let's discuss properly with the team. we can rename the table down the line with ease so we don't have to decide now.